client/clientservice/include/client/clientservice/configuration.hpp

```diff
@@ -26,6 +26,19 @@ void parseConfigFile(concord::client::concordclient::ConcordClientConfig&, const
 void configureSubscription(concord::client::concordclient::ConcordClientConfig&,
                            const std::string& tr_id,
                            bool insecure,
-                           const std::string& tls_path);
+                           const std::string& tls_path,
+                           const std::string& secrets_url);
+
+const std::string decryptPK(const std::optional<std::string>& secrets_url, const std::string& path);
```

Comment:
`decryptPk` (following the suggestions in the coding standard) or `decryptPrivateKey` or `decryptSecret`